src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java

```diff
@@ -15,5 +14,39 @@ public record CreateCollectionCommand(
         @Size(min = 1, max = 48)
         @Pattern(regexp = "[a-zA-Z][a-zA-Z0-9_]*")
         @Schema(description = "Name of the collection")
-        String name)
-    implements NamespaceCommand, NoOptionsCommand {}
+        String name,
+    @Nullable
+        @Schema(
+            description = "Configuration for the collection",
+            type = SchemaType.OBJECT,
+            implementation = Options.class)
+        Options options)
+    implements NamespaceCommand {
+  public record Options(
+
+      // limit of returned documents
+      @Schema(
+              description = "Vector search index configuration for the collection",
+              type = SchemaType.OBJECT,
+              implementation = VectorSearchConfig.class)
+          VectorSearchConfig vector) {
+
+    public record VectorSearchConfig(
+        @Schema(
+                description = "Vector field embedding size",
+                type = SchemaType.INTEGER,
+                implementation = Integer.class)
+            Integer size,
+        @Nullable
+            @Pattern(
+                regexp = "(dot_product|cosine|euclidean)",
+                message = "function name can only be 'dot_product', 'cosine' or 'euclidean'")
+            @Schema(
+                description =
+                    "Similarity function algorithm that needs to be used for vector search",
+                defaultValue = "cosine",
```

Comment:
Is "cosine" default that CQL has? (I think we want to use same as CQL default, if there is one)